packages/graphql/lib/src/link/error/link_error.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+
+import 'package:graphql/src/link/link.dart';
+import 'package:graphql/src/link/operation.dart';
+import 'package:graphql/src/link/fetch_result.dart';
+
+typedef ErrorHandler = void Function(ErrorResponse);
+
+class ErrorResponse {
```

Comment:
Might be better for this to use an [`OperationException`](https://github.com/zino-app/graphql-flutter/blob/602aa7fb01a8a433638c4ec3b8140c931bc4bb6a/packages/graphql/lib/src/exceptions/operation_exception.dart) for consistency

Reply:
Sure, I like the idea. Having said that, `ErrorResponse` also contains the relevant `Operation` and `FetchResult` (and also a `NextLink` for [retries](https://github.com/apollographql/apollo-link/blob/26b10d052bfcbc036deb0ff46b17c8872560873d/packages/apollo-link-error/README.md#retrying-failed-requests) but in hindsight, should be left out since I didn’t implement that).

Should I,

1. replace `ErrorResponse` with `OperationException` and not pass the `Operation` and `FetchResult`? 

2. replace `ErrorResponse` with `OperationException` but pass the `Operation` and `FetchResult` as callback parameters instead?

3. keep `ErrorResponse` but nest `OperationException` as a field?

Reply:
I'd nest the `OperationException`

Reply:
Done!

Reply:
Hi @jayjun I wonder why `NextLink` is left out. As in the linked apollo doc, I too facing issue with stale auth token, and like to retry the operation once get new token. Also, in the apollo implementation https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-error/src/index.ts , they don't do any special handling but just set the `NextLink` to `ErrorResponse`

Reply:
@bsr203 I left it out for the simple reason that I don’t use retries. Sounds like a bad reason but I feel if I don’t use it, I can’t do justice to the port.

Since you need it and know what changes to make, you should send a pull request.